src/appmixer/google/drive/NewFile/NewFile.js

```diff
@@ -29,6 +29,46 @@ const getNewFiles = async (lock, drive, folder, pageToken, newFiles = []) => {
     return { newFiles, newStartPageToken };
 };
 
+const detectNewFiles = async function(context) {
+
+    const { folder = {} } = context.properties;
+
+    let lock = null;
+    try {
+        lock = await context.lock(context.componentId, { maxRetryCount: 0 });
+    } catch (err) {
+        await context.stateSet('hasSkippedMessage', true);
+        return context.response();
+    }
+
+    try {
+        const { startPageToken, processedFiles = [] } = await context.loadState();
+        const auth = commons.getOauth2Client(context.auth);
+        const drive = google.drive({ version: 'v3', auth });
+
+        const { newFiles, newStartPageToken } = await getNewFiles(lock, drive, folder.id, startPageToken);
+
+        await context.stateSet('hasSkippedMessage', false);
```

Comment:
Should this be called before the line 49?
Scenario:
- webhook A arrives
- creates the lock and starts processing
- getNewFiles function finishes
- webhook B arrives (new file was just added)
- it detects the lock and sets `hasSkippedMessages`: true and quits
- the first process sets `hasSkippedMessages`: false
- the file that was supposed to be processed by webhook B is lost.